timeless/views.py

```diff
@@ -68,48 +68,115 @@ def render_template(self, context):
-                    form = forms.TableForm(request.form, instance=table)
-                else:
-                    form = forms.TableForm(request.form)
-                return render_template(
-                    "restaurants/tables/create_edit.html", form=form)
-
-            def post(self, pk=None, *args, **kwargs):
-                form = forms.TableForm(request.form)
-                if form.validate():
-                    form.save()
-                    return redirect(url_for("tables.list"))
-                return render_template(
-                    "restaurants/tables/create_edit.html", form=form)
+        class SettingsDetailView(views.DetailView):
+            model = models.ReservationSettings
+            template_name = "restaurants/tables/create_edit.html"
+            success_url_name = "reservation_settings_list"
+            not_found_url_name = "reservation_settings_list"
     """
-    def get_object(self):
-        """ Method for fetching object from db"""
-        raise NotImplementedError()
+    form = None
+    not_found_url_name = None
+    success_url_name = None
+
+    def get(self, id=None):
+        """
+        Get method fetch object from db and render it into template
+        """
+        assert self.form, "Form is required"
+        assert self.not_found_url_name, "not_found_url_name is required"
+
+        instance = self.get_object(id)
+        if not instance:
+            return redirect(url_for(self.not_found_url_name))
+
+        form = self.form.TableForm(request.form, instance=instance)
+        return self.render_template({"form": form})
+
+    def post(self, id=None):
+        """
+        Post method validate form from request and save it into db
+        """
+        form = self.form(request.form)
+        if form.validate():
+            form.save()
+            return redirect(url_for(self.success_url_name))
+        return self.render_template({"form": form})
+
+
+class CreateView(GenericView):
+    """Example
+        class SettingsCreateView(views.CreateView):
+            template_name = "restaurants/tables/create_edit.html"
+            success_url_name = "reservation_settings_list"
+            form = forms.SettingsForm
+    """
+    form = None
+    success_url_name = None
 
-    def get(self, *args, **kwargs):
-        """ Get method implementation """
-        raise NotImplementedError()
+    def get(self):
+        """Get method render create template """
+        assert self.form, "Form is required"
+        return self.render_template({"form": self.form()})
 
-    def post(self, *args, **kwargs):
-        """ Post method implementation"""
-        raise NotImplementedError()
+    def post(self):
+        """
+        Post method checks form validation,
+        save into db and redirect to success_url
+        """
+        assert self.form, "Form is required"
+        assert self.success_url_name, "Success_url_name is required"
```

Comment:
@vladarefiev these assertions seem like a primitive way of validating input data. What will happen if they fail? I suppose they will throw ``AssertionError`` and the user will see a stacktrace on the UI, which is not ok.

Reply:
@amihaiemil we can do it in this way
```
if not self.success_url_name:
    raise Exception
```
My point was that it's required values in that class.

Reply:
@vladarefiev the asserts are also fine as code, but I don't want the user to see any stacktrace on the UI. Is flask smart enough to only display the error message?

Reply:
@vladarefiev I don't want the user to see this stacktrace too. I want developer to see it during implementation. he should set these declarative variables otherwise he will se stacktrace in browser.

Reply:
@vladarefiev ok then, fine by me. Now there is a conflict, just fix it and we can merge :)